From 5e6966a80e787672454ac34e2dfd9790897b1e15 Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Mon, 11 Mar 2019 09:31:27 +0100
Subject: [PATCH] driver: net: mv88e6321: fix panic if probe deferred

If no irq is defined for mv88e6321 and probe will be
deferred, e.g. on mdio bus registration, driver
trys to free not registred irq and crashed. This patch
fix that.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/dsa/mv88e6xxx/chip.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/dsa/mv88e6xxx/chip.c b/drivers/net/dsa/mv88e6xxx/chip.c
index 7601255..6e1c642 100644
--- a/drivers/net/dsa/mv88e6xxx/chip.c
+++ b/drivers/net/dsa/mv88e6xxx/chip.c
@@ -4430,7 +4430,8 @@ static int mv88e6xxx_probe(struct mdio_device *mdiodev)
 out_remove_atu_flush:
 	device_remove_file(chip->dev, &dev_attr_atu_flush);
 out_g2_irq:
-	if (mv88e6xxx_has(chip, MV88E6XXX_FLAG_G2_INT))
+	if (mv88e6xxx_has(chip, MV88E6XXX_FLAG_G2_INT) &&
+	    chip->irq > 0)
 		mv88e6xxx_g2_irq_free(chip);
 out_g1_irq:
 	if (chip->irq > 0) {
-- 
2.7.4

